src/FormAlter/MediaFormAlter.php

```diff
@@ -0,0 +1,148 @@
+use Drupal\media_mpx\Repository\MpxMediaType;
+use Drupal\media_mpx\Service\UpdateVideoItem\UpdateVideoItem;
+use Drupal\media_mpx\Service\UpdateVideoItem\UpdateVideoItemRequest;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Alters the media form for mpx items to add "Reimport" button.
+ *
+ * @package Drupal\media_mpx\FormAlter
+ */
+class MediaFormAlter implements ContainerInjectionInterface {
+
+  use StringTranslationTrait;
+  use MessengerTrait;
+
+  /**
+   * The mpx media types repository.
+   *
+   * @var \Drupal\media_mpx\Repository\MpxMediaType
+   */
+  private $mpxMediaTypeRepository;
+
+  /**
+   * Update Video Item service.
+   *
+   * @var \Drupal\media_mpx\Service\UpdateVideoItem\UpdateVideoItem
+   */
+  private $updateService;
+
+  /**
+   * The custom media mpx logger service.
+   *
+   * @var \Drupal\media_mpx\MpxLogger
+   */
+  private $logger;
+
+  /**
+   * MediaFormAlter constructor.
+   */
+  private function __construct(MpxMediaType $mpxMediaTypeRepository, UpdateVideoItem $updateService, MpxLogger $logger) {
+    $this->mpxMediaTypeRepository = $mpxMediaTypeRepository;
+    $this->updateService = $updateService;
+    $this->logger = $logger;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('media_mpx.repository.mpx_media_types'),
+      $container->get('media_mpx.service.update_video_item'),
+      $container->get('media_mpx.exception_logger')
+    );
+  }
+
+  /**
+   * Implements hook_form_FORM_ID_alter().
+   */
+  public function alter(&$form, FormStateInterface $form_state, $form_id): void {
+    if ($this->alterAppliesToForm($form_state) === FALSE) {
+      return;
+    }
+
+    $form['actions']['reimport'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Reimport mpx data'),
+      '#submit' => [[$this, 'reimportCallback']],
+    ];
+  }
+
+  /**
+   * Callback for the 'reimport' button.
+   */
+  public function reimportCallback(array $form, FormStateInterface $formState) {
+    /* @var \Drupal\Core\Entity\ContentEntityForm $form_object */
+    $form_object = $formState->getFormObject();
+    $video = $form_object->getEntity();
+
+    $id_field = NULL;
+    if ($media_type = $this->mpxMediaTypeRepository->findByTypeId($video->bundle())) {
+      $field_map = $media_type->getFieldMap();
+      $id_field = $field_map['Media:id'] ?: NULL;
+    }
+
+    $mpx_id = $video->{$id_field}->value;
+    $this->updateVideoData((int) $mpx_id, $video->bundle());
+  }
+
+  /**
+   * Update video data and show success / error messages as relevant.
+   */
+  public function updateVideoData(int $mpxId, string $mediaTypeId) {
+    $request = new UpdateVideoItemRequest($mpxId, $mediaTypeId);
+    try {
+      $this->updateService->execute($request);
+      $this->messenger()->addMessage($this->t('The video has been reimported.'));
+    }
+    catch (\Exception $e) {
```

Comment:
Is there a more specific exception type to capture and report?

Reply:
@m4olivei No. The only possible ones are certainly well prevented before this point. This is just a catch all in case something completely unrelated breaks, to avoid a WSOD.

Reply:
Perfect. Normally this is bad, except this is UI code. 💯